packages/flutter_test/test/matchers_test.dart

```diff
@@ -173,6 +173,20 @@ void main() {
     expect('Foo#', isNot(equalsIgnoringHashCodes('Foo#00000')));
     expect('Foo#3421', isNot(equalsIgnoringHashCodes('Foo#00000')));
     expect('Foo#342193', isNot(equalsIgnoringHashCodes('Foo#00000')));
+    expect(<String>['Foo#a3b4d'], equalsIgnoringHashCodes(<String>['Foo#12345']));
+    expect(
+      <String>['Foo#a3b4d', 'Foo#12345'],
+      equalsIgnoringHashCodes(<String>['Foo#00000', 'Foo#00000']),
+    );
+    expect(
+      <String>['Foo#a3b4d', 'Bar#12345'],
+      equalsIgnoringHashCodes(<String>['Foo#00000', 'Bar#00000']),
+    );
+    expect(
+      <String>['Foo#a3b4d', 'Bar#12345'],
+      isNot(equalsIgnoringHashCodes(<String>['Bar#00000', 'Foo#00000'])),
+    );
+    expect(<String>['Foo#a3b4d'], isNot(equalsIgnoringHashCodes(<String>['Foo'])));
```

Comment:
Could you add some test cases showing that there will not be a match if the lists have different lengths?

Reply:
Done